internal/jar/jar.go

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2022.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jar
+
+import (
+	"archive/zip"
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+)
+
+const (
+	manifestPath = "META-INF/MANIFEST.MF"
+)
+
+func GetManifest(jar string) (*Manifest, error) {
+
+	zipfile, err := zip.OpenReader(jar)
+	if err != nil {
+		return nil, err
+	}
+	defer zipfile.Close()
+
+	for _, file := range zipfile.File {
+		// check if the file matches the name for application portfolio xml
+		if file.Name == manifestPath {
+			reader, err := file.Open()
+			if err != nil {
+				return nil, err
+			}
+			defer reader.Close()
+
+			if manifestMap, err := parseManifest(reader); err != nil {
```

Comment:
Make `parseManifest` return `(*Manifest, error)` so you can just `return parseManifest(reader)` here?